contracts/Permissions.sol

```diff
@@ -432,4 +403,25 @@ library Permissions {
     function unpackParamCount(uint256 config) internal pure returns (uint8) {
         return uint8(config >> 248);
     }
+
+    function keyForFunctions(address targetAddress, bytes4 functionSig)
+        public
+        pure
+        returns (bytes32)
+    {
+        // fits in 32 bytes
+        return bytes32(abi.encodePacked(targetAddress, functionSig));
+    }
+
+    function keyForCompValues(
+        address targetAddress,
+        bytes4 functionSig,
+        uint8 paramIndex
+    ) public pure returns (bytes32) {
+        // fits in 32 bytes
+        return
+            bytes32(
+                abi.encodePacked(targetAddress, functionSig, ":", paramIndex)
```

Comment:
Why is `":"` necessary?

Reply:
This was a KISS approach. Fears of key clashing for paramIndex 0, with the other key generator function. But they target different maps. Removing, thanks

Reply:
encodePacked also forces the width of a value to be specific to the type, meaning leading 0s are not removed. So if you use only static type you should not have any issues.